### PR TITLE
Deprecate gui qt

### PR DESCRIPTION
**Pull request by tlambert03**
# Description
I think it's time to deprecate gui_qt 😱
Most people new to napari are still learning it with `gui_qt`... and I don't see much reason to keep that up.  This doesn't break anything... just formally deprecates it and updates all of the examples.

I'll follow up with an updated event_loop docs, and some PRs to napari tutorials / aka napari.github.io

## Type of change

- [x]  Deprecation

# References

# How has this been tested?

- [ ] example: the test suite for my feature covers cases x, y, and z
- [ ] example: all tests pass with my change

## Final checklist:
- [ ] My PR is the minimum possible work for the desired functionality
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] If I included new strings, I have used `trans.` to make them localizable.
      For more information see our [translations guide](https://napari.org/docs/dev/guides/translations.html).

### CODE REVIEW

File: napari/_qt/qt_event_loop.py
Comment (sofroniewn) [inline napari/_qt/qt_event_loop.py:241]: Should we say "and will be removed in `0.5.0`"? Seems appropriate to leave this around until then

**Review by sofroniewn — APPROVED** [review]
Looks good to me, but let's let more people weigh in before merge!! We should probably time the merge of this PR with the merge of https://github.com/napari/napari.github.io/pull/124 too which will remove its reference from our docs too. I think at that point we'll be good!!

File: napari/_qt/qt_event_loop.py
Comment (jni) [reply napari/_qt/qt_event_loop.py:241]: I think @tlambert03 left that out because I suggested in the past that we should have a longer deprecation cycle for this, since so many people will be using this API... If we want to set a release on it I would suggest 0.6. What do you think @sofroniewn?

**Comment by jni** [comment]
I should say, I'm A-ok with this change, thanks for pushing us forward @tlambert03!

File: napari/_qt/qt_event_loop.py
Comment (sofroniewn) [reply napari/_qt/qt_event_loop.py:241]: `0.6.0` is fine with me, though it feels a long way out! I don't see that maintaining this API is causing us problems right now, so ok to have a long drop time. @tlambert03 you ok with this?

Comment (tlambert03) [reply napari/_qt/qt_event_loop.py:241]: other than _really_ wanting people not to use it, I don't have any specific objection :).  I'd almost rather leave it unspecified (so we can go more by the number of months and not some ill defined future point).

Comment (sofroniewn) [reply napari/_qt/qt_event_loop.py:241]: yeah, the uncertainty around what and when `0.6.0` would even be isn't ideal for me, but i think specifying could help people feel less taken by surprise when it does go. i'll defer to @jni final opinion here

Comment (jni) [reply napari/_qt/qt_event_loop.py:241]: I think we can defer the precise timing of the removal to a future PR. When it does come time to remove it, we can have a final two-patch-release notice. In the meantime I'm just going to merge this as-is! 🎉

Comment (tlambert03) [reply napari/_qt/qt_event_loop.py:241]: Thank you for your service `gui_qt()`! ❤️

**Pull request merged**
